docs/questions/error-recovery-best-practices.md

```diff
@@ -0,0 +1,95 @@
+### Question
+
+
+What are the best practices for error recovery in chaincode?
```

Comment:
I have a feeling both files: `docs/questions/error-recovery-best-practices.md` and `docs/questions/errors-and-exceptions-in-chaincode.md` are mostly irrelevant and misleading. What do we need is:

1. The explanation of our error classes in general (the hierarchy, what is error key, what is error code, common methods). I'm not sure if Q&A is a good place, maybe it would be better for concepts. https://github.com/GalaChain/sdk/blob/main/chain-api/src/utils/error.ts
2. The explanation how errors handling should look like in the transaction - uncaught throw => transaction fails, nothing is saved, and it it guaranteed here: https://github.com/GalaChain/sdk/blob/main/chaincode/src/contracts/GalaTransaction.ts#L232
3. Edge cases, when partial success is allowed, like in `BatchSubmit`